packages/sdk/src/interaction/interaction.ts

```diff
@@ -104,79 +85,88 @@ const IX = `{
+export const isArray = (d: any): d is any[] => Array.isArray(d)
+export const isObj = (d: any): d is Record<string, any> => d !== null && typeof d === "object"
+export const isNull = (d: any): d is null => d == null
+export const isFn = (d: any): d is Function => typeof d === "function"
 
-export const isInteraction = ix => {
+export const isInteraction = (ix: IIx) => {
   if (!isObj(ix) || isNull(ix) || isNumber(ix)) return false
   for (let key of KEYS) if (!ix.hasOwnProperty(key)) return false
   return true
 }
 
-export const Ok = ix => {
+export const Ok = (ix: IIx) => {
   ix.status = OK
   return ix
 }
 
-export const Bad = (ix, reason) => {
+export const Bad = (ix: IIx, reason: string) => {
   ix.status = BAD
   ix.reason = reason
   return ix
 }
 
-const makeIx = wat => ix => {
+const makeIx = (wat: string) => (ix: IIx) => {
   ix.tag = wat
   return Ok(ix)
 }
 
-const prepAccountKeyId = acct => {
+const prepAccountKeyId = (acct: Partial<IAcct> | IAcctFn): Partial<IAcct> | IAcctFn => {
   if (acct.keyId == null) return acct
 
-  invariant(!isNaN(parseInt(acct.keyId)), "account.keyId must be an integer")
+  invariant(!isNaN(parseInt(acct.keyId.toString())), "account.keyId must be an integer")
+
   return {
     ...acct,
-    keyId: parseInt(acct.keyId),
-  }
+    keyId: parseInt(acct.keyId.toString()),
+  } as IAcct | IAcctFn
+}
+
+interface IPrepAccountOpts {
+  role?: string | null
 }
 
-export const prepAccount = (acct, opts = {}) => ix => {
+export const initAccount = (): IAcct => JSON.parse(ACCT)
+
+export const prepAccount = (acct: IAcct | IAcctFn, opts: IPrepAccountOpts = {}) => (ix: IIx) => {
   invariant(
     typeof acct === "function" || typeof acct === "object",
     "prepAccount must be passed an authorization function or an account object"
   )
   invariant(opts.role != null, "Account must have a role")
 
-  const ACCOUNT = JSON.parse(ACCT)
-  const role = opts.role
+  const ACCOUNT = initAccount()
+  const role = opts.role as (typeof AUTHORIZER | typeof PAYER | typeof PROPOSER)
   const tempId = uuid()
+  let account: Partial<IAcct> = acct
 
   if (acct.authorization && isFn(acct.authorization))
-    acct = {resolve: acct.authorization}
-  if (!acct.authorization && isFn(acct)) acct = {resolve: acct}
+    account = {resolve: acct.authorization}
+  if (!acct.authorization && isFn(acct)) account = {resolve: acct}
 
-  const resolve = acct.resolve
-  if (resolve)
-    acct.resolve = (acct, ...rest) =>
+  const resolve = account.resolve
+  if (resolve) {
+    account.resolve = (acct: IAcct, ...rest: any[]) =>
       [resolve, prepAccountKeyId].reduce(
         async (d, fn) => fn(await d, ...rest),
         acct
       )
-  acct = prepAccountKeyId(acct)
+  }
+  account = prepAccountKeyId(account)
 
   ix.accounts[tempId] = {
     ...ACCOUNT,
     tempId,
-    ...acct,
+    ...account,
     role: {
       ...ACCOUNT.role,
       ...(typeof acct.role === "object" ? acct.role : {}),
```

Comment:
```suggestion
      ...(typeof account.role === "object" ? account.role : {}),
```
If we want to create a new variable `account` in lieu of `acct` & preserve the initial behaviour, then shouldn't this reference be replaced as well?